packages/playwright-core/src/client/errors.ts

```diff
@@ -35,6 +35,21 @@ export function isTargetClosedError(error: Error) {
   return error instanceof TargetClosedError;
 }
 
+export class OverriddenAPIError extends Error {
+  public apiNameOverride: string;
+
+  constructor(error: Exclude<SerializedError['error'], undefined>, apiNameOverride: string) {
+    super(error.message);
```

Comment:
We already have 4 layers of processing of the api names and errors, adding another one for this use case will add to the confusion.